src/server/pfs/server/compaction.go

```diff
@@ -111,9 +125,11 @@ func (c *compactor) createCompactTasks(ctx context.Context, taskDoer task.Doer,
 	}
 	var tasks []*CompactTask
 	var taskLens []int
-	for _, res := range results {
+	for i := 0; i < len(results); i += len(shards) {
```

Comment:
it's not really the order in which they are returned (which, as you say, is not guaranteed), since the code generating the tasks and sending them to the `Doer` can handle the output however it wants - in particular we record the index, so the usual pattern is slotting them into an array in the same order

Reply:
The task service knows the order that the tasks came in and will report that through the index in the collection callback, which the caller can use to insert the results in the correct position (usually a results slice).